spring-cloud-stream-binders/spring-cloud-stream-binder-redis/src/main/java/org/springframework/cloud/stream/binder/redis/RedisMessageChannelBinder.java

```diff
@@ -165,6 +165,8 @@ public RedisMessageChannelBinder(RedisConnectionFactory connectionFactory,
 	@Override
 	protected void onInit() {
 		this.errorAdapter.setIntegrationEvaluationContext(this.evaluationContext);
+		this.errorAdapter.setBeanFactory(getBeanFactory());
+		this.errorAdapter.afterPropertiesSet();
```

Comment:
Looking at how the `errorAdapter` is setup, I think we'd best move its entire initialization in `onInit()` - makes it an easier read than the current split.